DOC Add dropdowns in Module 3.3

manuel-morales-a opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Towards #26617.


#### What does this implement/fix? Explain your changes.

This PR implements the dropdowns in section 3.3, using the resources of #26625. The only file that had to been modified is `model_evaluation.rst`.

Thank you!


#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

manuel-morales-a committed:
Subsection 3.3.1 and 3.3.2 ready

manuel-morales-a committed:
Subsection 3.3.3 ready

manuel-morales-a committed:
Subsection 3.3.4 ready

ArturoAmorQ commented on doc/modules/model_evaluation.rst:
```diff
@@ -2317,10 +2355,16 @@ function::
     for an example of mean squared error usage to
     evaluate gradient boosting regression.
 
+|details-start|
+**Square root of the MSE**
+|details-split|
+
 Taking the square root of the MSE, called the root mean squared error (RMSE), is another
 common metric that provides a measure in the same units as the target variable. RSME is
 available through the :func:`root_mean_squared_error` function.
 
+|details-end|
+
```
This paragraph is short enough not to be hidden in a dropdown. It is also a common use metric, so we rather keep it visible.

manuel-morales-a replied:
Very good! It has been de-dropdowned in one of the following commits.

ArturoAmorQ reviewed (commented):
Thanks for the PR @manuel-morales-a! Now that you're working on this section, can you also please hide
- the `One-vs-one Algorithm` and `One-vs-rest Algorithm` of the [Multi-class case subsection in 3.3.2.15.](https://scikit-learn.org/stable/modules/model_evaluation.html#multi-class-case) (in separate dropdowns)
- the `Interpretation across varying prevalence` and `Mathematical divergences` in [3.3.2.19. Class likelihood ratios](https://scikit-learn.org/stable/modules/model_evaluation.html#class-likelihood-ratios) (also in separate dropdowns)
- the subsections of [3.3.4.11. D² score](https://scikit-learn.org/stable/modules/model_evaluation.html#d2-score) (one dropdown per deviance)

manuel-morales-a committed:
Fixed 3.3.2.15

manuel-morales-a committed:
Fixed 3.3.2.19

manuel-morales-a committed:
Fixed 3.3.4.11

manuel-morales-a committed:
Fixed square root of MSE message

manuel-morales-a commented:
Thanks for the feedback @ArturoAmorQ! I've implemented your comments. 

I have one remaining question. In [3.3.2.19. Class likelihood ratios](https://scikit-learn.org/stable/modules/model_evaluation.html#class-likelihood-ratios), with the dropdowns I've implemented, the `Examples` section ends up surrounded by dropdowns (between `Mathematical divergences` and `References`). Should I move it above the `Interpretation across varying prevalence` dropdown, next to the main text, or simply leave it where it currently is?

Thank you!

ArturoAmorQ commented:
> Should I move it above the `Interpretation across varying prevalence` dropdown, next to the main text, or simply leave it where it currently is?

I would rather move it above the `Interpretation across varying prevalence` dropdown, please :)

manuel-morales-a committed:
Moved example nearer to main text

manuel-morales-a commented:
Thank you! It should be good to go now: moved the examples as discussed. Cheers

ArturoAmorQ commented on doc/modules/model_evaluation.rst:
```diff
@@ -580,7 +580,9 @@ or *informedness*.
       is computed for each class and then averaged over total number of classes.
 
-.. topic:: References:
+|details-start|
+**References**
+|details-split|
```
Sorry just a last comment, I just noticed some of the references (those with the format `[authorDATE]`) are called in some other parts of this page, but as of today, dropdowns break the sphynx cross-referencing system. Can you please revert the reference back to a `topic` directive whenever at least one of the references are called elsewhere?

manuel-morales-a replied:
No problem! I've reinstated the topic directives when needed in the commit below. I had to remove a couple of dropdowns but most remained

ArturoAmorQ commented on doc/modules/model_evaluation.rst:
```diff
@@ -1503,11 +1492,7 @@ the keyword argument ``multiclass`` to ``'ovo'`` and ``average`` to
-|details-start|
-**One-vs-rest Algorithm**
-|details-split|
+.. topic:: One-vs-rest Algorithm:
```
I am so sorry! I may have not been clear enough :sweat_smile:

The references get broken whenever they point inside a dropdown. Cross-references from the dropdown to the outside (such as this one) are perfectly fine!

The underlying issue is that the dropdown has to be unfolded for mozilla to access the reference.

manuel-morales-a replied:
No probs! I've updated the references to topic directives when required (assuming I understood correctly this time. If not, no problem and we'll keep modifying the rst)

manuel-morales-a committed:
Updating references to topic directive when needed

ArturoAmorQ reviewed (approved):
LGTM! Thanks again for your work and patience and congrats on your first scikit-learn merge :) Hope this is just the first one from a series of contributions!

manuel-morales-a commented:
Thanks for the helpful tips and comments, Arturo! Talk to you soon in another contribution
